dirty_equals/_other.py

```diff
@@ -145,3 +146,40 @@ def is_even(x):
 
     def equals(self, other: Any) -> bool:
         return self.func(other)
+
+
+class IsHash(DirtyEquals[str]):
+    """
+    A class that checks if a value is a valid common hash type, using a simple length and allowed characters regex.
+    """
+
+    def __init__(self, hash_type: Literal['md5', 'sha-1', 'sha-256']):
+        """
+        Args:
+            hash_type: The hash type to check. Must be specified.
+
+        ```py title="IsUUID"
+        from dirty_equals import IsHash
+
+        assert 'f1e069787ece74531d112559945c6871' == IsHash('md5')
+        assert 'f1e069787ece74531d112559945c6871' != IsHash('sha-256')
+        assert 'F1E069787ECE74531D112559945C6871' == IsHash('md5')
+        assert '40bd001563085fc35165329ea1ff5c5ecbdbbeef' == IsHash('sha-1')
+        assert 'a665a45920422f9d417e4867efdc4fb8a04a1f3fff1fa07e998e86f7f7a27ae3' == IsHash('sha-256')
+        ```
+        """
+
+        allowed_hashes = 'md5', 'sha-1', 'sha-256'
+        if hash_type not in allowed_hashes:
```

Comment:
```suggestion
        if hash_type not in HashTypes.__args__:
```

Reply:
That's neat! Didn't know you could do that